### PR TITLE
Add request and publish_request

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/src/lib.rs [hunk 1]
```diff
@@ -129,6 +131,7 @@ pub enum ClientOp {
     Publish {
         subject: String,
         payload: Bytes,
+        respond: Option<String>,
```
Comment (caspervonb) [inline async-nats/src/lib.rs:134]: For naming parity with other clients, `reply_to`

**Comment by caspervonb** [comment]
Bit of name bike shedding again.
We need to decide how we want to deal with the overload, e.g `publish_with_<overload>`, `publish_opt(opts: PublishOptions)` etc?

**Comment by Jarema** [comment]
@caspervonb  I think that `{whatever/publish}_with_options` is best.
Does not affect `publish_request` IMO as that is literally publishing a request 🙂 .

**Comment by caspervonb** [comment]
Could also be something like `publish_with_reply`, are we gonna do `publish_with_headers`?

**Review by caspervonb — APPROVED** [review]
Internal fields in ops could have some parity but lgtm.

**Pull request merged**